Include/object.h

```diff
@@ -283,6 +283,19 @@ _Py_ThreadId(void)
     // Both GCC and Clang have supported __builtin_thread_pointer
     // for s390 from long time ago.
     tid = (uintptr_t)__builtin_thread_pointer();
+#elif defined(thread_local) || defined(__GNUC__)
+    #if defined(thread_local)
+    static thread_local int __tp = 0;
```

Comment:
I don't think this will do what we want.`__tp` will be a unique variable in each C file that includes this header.

Reply:
To expand on this: if you declare a `static` variable inside a `static inline` function, then that variable is different in each translation unit (C file) that includes the `static inline` function.